### DISCUSSION TITLE
Repatriation - Website How-to

### DISCUSSION THREAD

**Discussion by magpiedin** [fieldmuseum/EMu-Documentation#1003] [HOW TO...]
### Documentation related to https://repatriation.fieldmuseum.org
  - External **[PDF here](https://github.com/fieldmuseum/EMu-Documentation/blob/master/Workflows/FMNH_Repatriation_Website_HowTo.pdf)** [2025]
  - Internal [Google Doc here](https://docs.google.com/document/d/1CAhlaIHdxNk1hihmMqugdLu2JNhkAfo162SK4ACNdmg)

---
### EMu Exports - background and setup:

**How to do a manual data refresh**
1. Search for all consultation records with RP-6725 attached
2. Run the CCNAHRS_Repat_sensitive export on ALL records
3. After the exports have finished, contact IT because they will have to do a manual import of the data

<img width="414" alt="Export_Repat" src="https://github.com/user-attachments/assets/9c26d8f6-4c8a-46f9-924d-4252624d6ee7" />

The Record Count should equal the number of consultation records.

<img width="473" alt="Repat_Consultations" src="https://github.com/user-attachments/assets/632da04c-0028-4170-8891-b0b46c66dc72" />

**Security**
If access to all records that come back in the private search needs to be shut off:
- Shut off the Consultations “CCNAHR_Repat_sensitive” Export 
    1. Log into EMu as user “emu”
    2. In Consultations module, retrieve a few records, and go to Tools → Exports
      <img width="417" alt="repat_export_consults" src="https://github.com/user-attachments/assets/7c1c4462-bdce-456c-b49e-37c51ebd2b9d" />

    3. Edit properties for the CCNAHR_Repat_sensitive
      - Enter the current date in the “Finish Date” field and press OK
      <img width="328" alt="repat_export_prop" src="https://github.com/user-attachments/assets/c6da84aa-4b9d-4d54-be2f-3968c02c8e3a" />

**Comment by peteherbst** [comment]
### Processing the exported EMu XML file

The XML file goes through a simple ETL process (in our case PHP Laravel [Artisan Console](https://laravel.com/docs/8.x/artisan) commands) to insert the required data into a local SQLite database file on scheduled basis using Laravel [Cron scheduling](https://laravel.com/docs/8.x/scheduling). If you'd like more details on the data-end of things regarding how the data is processed, grouped, etc., contact Kate Webbink and Pete Herbst for details.

### Serving the data from SQLite

We use a combination of [datatables](https://datatables.net/) and [Chart.js](https://www.chartjs.org/) to display the processed information on webpages.
